passive/SQL injection detection.js

```diff
@@ -1,383 +1,89 @@
-      wascId[0],
-      msg
-    );
-  }
-  if (sqlite.test(body)) {
-    sqlite.lastIndex = 0;
-    var foundsqlite = [];
-    while ((comm = sqlite.exec(body))) {
-      foundsqlite.push(comm[0]);
-    }
-    ps.raiseAlert(
-      alertRisk[3],
-      alertConfidence[1],
-      alertTitle[8],
-      alertDesc[8],
-      url,
-      "",
-      "",
-      foundsqlite.toString(),
-      alertSolution[0],
-      "",
-      cweId[0],
-      wascId[0],
-      msg
-    );
-  }
-  if (sapdb.test(body)) {
-    sapdb.lastIndex = 0;
-    var foundsapdb = [];
-    while ((comm = sapdb.exec(body))) {
-      foundsapdb.push(comm[0]);
-    }
-    ps.raiseAlert(
-      alertRisk[3],
-      alertConfidence[2],
-      alertTitle[9],
-      alertDesc[9],
-      url,
-      "",
-      "",
-      foundsapdb.toString(),
-      alertSolution[0],
-      "",
-      cweId[0],
-      wascId[0],
-      msg
-    );
-  }
-  if (sybase.test(body)) {
-    sybase.lastIndex = 0;
-    var foundsybase = [];
-    while ((comm = sybase.exec(body))) {
-      foundsybase.push(comm[0]);
-    }
-    ps.raiseAlert(
-      alertRisk[3],
-      alertConfidence[2],
-      alertTitle[10],
-      alertDesc[10],
-      url,
-      "",
-      "",
-      foundsybase.toString(),
-      alertSolution[0],
-      "",
-      cweId[0],
-      wascId[0],
-      msg
-    );
-  }
-  if (ingress.test(body)) {
-    ingress.lastIndex = 0;
-    var foundingress = [];
-    while ((comm = ingress.exec(body))) {
-      foundingress.push(comm[0]);
-    }
-    ps.raiseAlert(
-      alertRisk[3],
-      alertConfidence[2],
-      alertTitle[11],
-      alertDesc[11],
-      url,
-      "",
-      "",
-      foundingress.toString(),
-      alertSolution[0],
-      "",
-      cweId[0],
-      wascId[0],
-      msg
-    );
-  }
-  if (frontbase.test(body)) {
-    frontbase.lastIndex = 0;
-    var foundfrontbase = [];
-    while ((comm = frontbase.exec(body))) {
-      foundfrontbase.push(comm[0]);
-    }
-    ps.raiseAlert(
-      alertRisk[3],
-      alertConfidence[2],
-      alertTitle[12],
-      alertDesc[12],
-      url,
-      "",
-      "",
-      foundfrontbase.toString(),
-      alertSolution[0],
-      "",
-      cweId[0],
-      wascId[0],
-      msg
-    );
-  }
-  if (hsqldb.test(body)) {
-    hsqldb.lastIndex = 0;
-    var foundhsqldb = [];
-    while ((comm = hsqldb.exec(body))) {
-      foundhsqldb.push(comm[0]);
+  const body = msg.getResponseBody().toString();
+  for (let i = 0; i < sqlImplementations.length; i++) {
+    const sqlImpl = sqlImplementations[i];
+    const alertTitle = `Information Disclosure - ${sqlImpl.name} error`;
+    const article = sqlImpl.name.match(/^(?:[IO]|MS|SQ|H)/) ? "An" : "A";
+    const alertDesc = `${article} ${sqlImpl.name} error was discovered in the HTTP response body.`;
+    if (sqlImpl.regex.test(body)) {
+      sqlImpl.regex.lastIndex = 0;
+      const found = [];
+      let comm;
+      while ((comm = sqlImpl.regex.exec(body))) {
+        found.push(comm[0]);
```

Comment:
Did this come from one of the comment scripts?

Reply:
If so, think it's worth renaming more meaningfully?

Reply:
I combined all the individual checks into a single loop. Do you mean renaming the `found` variable? Maybe `foundSqlErrors`?

Reply:
Sorry I meant the “comm” array

Reply:
Ugh I didn't realize how many times this had been copied. Anyway thanks for fixing this one. It seems that people have repeated use of comm[] all over the place :wink: